randomwallpaper@iflow.space/elements.js

```diff
@@ -97,17 +100,46 @@ var HistoryElement = GObject.registerClass({
 			this.menu.addMenuItem(new PopupMenu.PopupMenuItem('Unknown source.'));
 		}
 
+		this.copyToFavorites = new PopupMenu.PopupMenuItem('Save for later');
+		this.copyToFavorites.connect('activate', () => {
+			let sourceFile = Gio.File.new_for_path(this.historyEntry.path);
+			let targetFolder = Gio.File.new_for_path(this._settings.get('favorites-folder', 'string'));
+			let targetFile = targetFolder.get_child(historyEntry.name);
+
+			try {
+				if (!targetFolder.make_directory_with_parents(null)) {
+					this.logger.warning('Could not create directories.');
+					return;
+				}
+			} catch (error) {
+				if (error === Gio.IOErrorEnum.EXISTS) { }
```

Comment:
I also felt for this one at first.

Despite the docs saying otherwise this throws an error instead of returning `false` so this branch is not unnecessary? I should probably invert it. I do not know when what is thrown or returned exactly.

![grafik](https://user-images.githubusercontent.com/1408843/221387355-f9761ea7-7434-4191-8999-0c40687b36de.png)

Reply:
Alright!